typewriter/fixes/fixer_utils.py

```diff
@@ -133,3 +145,207 @@ def touch_import(package, name, node):
 
     create_import(package, name, node)
     return name
+
+
+_block_syms = {syms.funcdef, syms.classdef, syms.trailer}
+
+
+def _find(name, node):
+    nodes = [node]
+    while nodes:
+        node = nodes.pop()
+        if node.type > 256 and node.type not in _block_syms:
+            nodes.extend(node.children)
+        elif node.type == token.NAME and node.value.strip() == name:
+            return node
+    return None
+
+
+ImportPairing = NamedTuple("ImportPairing", [
+    ("entry", str),
+    ("binding_name", str)
+])
+
+ImportNodeInfo = NamedTuple("ImportNodeInfo", [
+    ("imports", Dict[str, Set[ImportPairing]]),
+    ("node", Node)
+])
+
+ImportInfo = NamedTuple("ImportInfo", [
+    ("entry", str),
+    ("package", str),
+    ("binding", str)
+])
+
+
+def find_import_info(package, name, node):
+    # type: (str, str, Node) -> Optional[ImportInfo]
+    """
+    Finds the import statement for <name> regardless of whether <name> is the binding name
+        ex where name='a':
+        import a => match AND import a as b => match BUT import b as a => None
+
+    Parameters
+    -----------
+    package : str
+    name : str
+    node : Node
+
+    Return
+    -----------
+    Optional[ImportInfo]
+    """
+    for child in node.children:
+        if is_import(child):
+            ret = get_import_info(child)
+            imports = ret.imports.get(package)
+            if imports is None:
+                return None
+            import_ = next((x for x in imports if x.entry == name), None)
+            if import_:
+                return ImportInfo(import_.entry, package, import_.binding_name)
+        elif child.type == syms.simple_stmt:
+            res = find_import_info(package, name, child)
+            if res is None:
+                continue
```

Comment:
Ah, nice catch here.  Out of curiosity, was this exposed by the new test, or was this an independent finding?